transport/swift/swift_test.go

```diff
@@ -0,0 +1,136 @@
+package swift
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/imgproxy/imgproxy/v3/config"
+	"github.com/ncw/swift/v2"
+	"github.com/ncw/swift/v2/swifttest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	testContainer = "test"
+	testObject    = "test.png"
+)
+
+type SwiftTestSuite struct {
+	suite.Suite
+	server *swifttest.SwiftServer
+}
+
+func (s *SwiftTestSuite) SetupSuite() {
+	s.server, _ = swifttest.NewSwiftServer("localhost")
+
+	config.Reset()
+
+	config.SwiftAuthURL = s.server.AuthURL
+	config.SwiftUsername = swifttest.TEST_ACCOUNT
+	config.SwiftAPIKey = swifttest.TEST_ACCOUNT
+	config.SwiftAuthVersion = 1
+
+	s.setupTestFile()
+}
+
+func (s *SwiftTestSuite) setupTestFile() {
+	t := s.T()
+	c := &swift.Connection{
+		UserName:    config.SwiftUsername,
+		ApiKey:      config.SwiftAPIKey,
+		AuthUrl:     config.SwiftAuthURL,
+		AuthVersion: config.SwiftAuthVersion,
+	}
+
+	ctx := context.Background()
+
+	err := c.Authenticate(ctx)
+	assert.Nil(t, err, "failed to authenticate with test server")
+
+	err = c.ContainerCreate(ctx, testContainer, nil)
+	assert.Nil(t, err, "failed to create container")
+
+	f, err := c.ObjectCreate(ctx, testContainer, testObject, true, "", "image/png", nil)
+	assert.Nil(t, err, "failed to create object")
+
+	defer f.Close()
+
+	wd, err := os.Getwd()
+	assert.Nil(t, err)
+
+	data, err := ioutil.ReadFile(filepath.Join(wd, "..", "..", "testdata", "test1.png"))
+	assert.Nil(t, err, "failed to read testdata/test1.png")
+
+	b, err := f.Write(data)
+	assert.Greater(t, b, 100)
+	assert.Nil(t, err)
+}
+
+func (s *SwiftTestSuite) TearDownSuite() {
+	s.server.Close()
+}
+
+func (s *SwiftTestSuite) TestNew() {
+	transport, err := New()
+
+	assert.Nil(s.T(), err, "failed to set up transport")
+	assert.IsType(s.T(), transport, transport)
+}
+
+func (s *SwiftTestSuite) TestRoundTripWithETagDisabledReturns200() {
+	config.ETagEnabled = false
+	request, _ := http.NewRequest("GET", "swift:///test/test.png", nil)
+
+	transport, _ := New()
```

Comment:
It looks like we can initialize transport in `SetupSuite` as it doesn't change between tests.